src/message_metadata.cc

```diff
@@ -0,0 +1,124 @@
+//---------------------------------------------------------------------
+//---------------------------------------------------------------------
+#include "message_metadata.h"
+#include "lv_interop.h"
+#include "well_known_types.h"
+
+namespace grpc_labview
+{
+    //---------------------------------------------------------------------
+    // Functions for calculating whether an element is considered a well known type.
+    //---------------------------------------------------------------------
+    static wellknown::Types IsWellKnownDouble2DArray(const MessageElementMetadata& metadata)
```

Comment:
IsWellKnownDouble2DArray sounds like a predicate that returns bool. I recommend renaming this to something like GetDouble2DArrayWellKnownTypeEnum or converting it to a lambda so you can include it in the map initializer.

e.g. 
```
[](const MessageElementMetadata& metadata) {
        if (!metadata.isRepeated)
        {
            return wellknown::Types::Double2DArray;
        }
        return wellknown::Types::None;
}
```

Reply:
Updated to `CalculateDouble2DArrayWellKnownTypeEnum`.